src/Expander.hs

```diff
@@ -287,7 +288,10 @@ evalMod (Expanded em _) = execWriterT $ do
             Map.insert cn (ConstructorInfo argTypes dt)
           lift $ modifyState $
             over (expanderWorld . worldDatatypes . at p . non Map.empty) $
-            Map.insert dt (DatatypeInfo arity [c | (_, c, _) <- ctors ])
+            Map.insert dt $ DatatypeInfo
+              { _datatypeArgKinds     = replicate (fromIntegral arity) KStar
```

Comment:
this `arity` comes from the `Data` constructor above; I guess the next step will be to push the `[Kind]` representation a bit further and use it in `Data` as well.

Reply:
done.